### PR TITLE
Refs #29548 -- Fixed GIS tests on MariaDB

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

**Comment by claudep** [comment]
buildbot, test on mariadb.

File: django/contrib/gis/db/backends/mysql/introspection.py [hunk 1]
```diff
@@ -29,9 +29,12 @@ def get_geometry_type(self, table_name, geo_col):
...
 
     def supports_spatial_index(self, cursor, table_name):
-        # Supported with MyISAM, or InnoDB on MySQL 5.7.5+
+        # Supported with MyISAM/Aria, or InnoDB on MySQL 5.7.5+/MariaDB 10.2.2+
         storage_engine = self.get_storage_engine(cursor, table_name)
         return (
-            (storage_engine == 'InnoDB' and self.connection.mysql_version >= (5, 7, 5)) or
-            storage_engine == 'MyISAM'
+            storage_engine in ['MyISAM', 'Aria'] or
+            (storage_engine == 'InnoDB' and not self.connection.mysql_is_mariadb and
```
Comment (timgraham) [inline django/contrib/gis/db/backends/mysql/introspection.py]: How about:
```
if storage_engine == 'InnoDB':
    return self.connection.mysql_version >= (
        (10, 2, 2) if self.connection.mysql_is_mariadb else (5, 7, 5)
    )
    
return storage_engine in ('MyISAM', 'Aria')
```

File: tests/gis_tests/inspectapp/tests.py [hunk 2]
```diff
@@ -141,8 +141,10 @@ def test_time_field(self):
         else:
             self.assertIn('    f_decimal = models.DecimalField(max_digits=0, decimal_places=0)', model_def)
         self.assertIn('    f_int = models.IntegerField()', model_def)
-        self.assertIn('    f_datetime = models.DateTimeField()', model_def)
-        self.assertIn('    f_time = models.TimeField()', model_def)
+        if connection.vendor != 'mysql' or connection.mysql_is_mariadb:
```
Comment (timgraham) [inline tests/gis_tests/inspectapp/tests.py]: I'm not sure if this skipping logic is correct -- it seems to skip on MySQL but not MariaDB.

**Comment by claudep** [comment]
buildbot, test on mariadb.

**Comment by claudep** [comment]
buildbot, test on mariadb.

**Pull request merged**